.gitignore

```diff
@@ -19,3 +19,4 @@ etl/logs
 etl/test/**/logs
 /etl/data_management/upload/
 /etl/data_management/backups/
+/etl/data_management/upload_origin.txt
```

Comment:
> 
> 
> The behaviour is as described.
> 
> To the wiki changes:
> 
>     * `If data exists`: What happens if no data exist?
The step is left out, as indicated by the 'if'. Added clarification to the updated version.
>     * `limit is set`: What is currently the limit. Many this is helpful to know without looking into the code.
I disagree, it might get changed later and if it does, the wiki will probably not changed accordingly. We already have proof for this in the wiki, for example, the overview was already out of date and stated that only the ehrenamtssuche crawler exists. 
>     * Currently, it looks like there are still incorrect line breaks in the proposal.
If you are referring to the blank lines, those are formatting as a block of text is unpleasant to read. Otherwise it is due to the way github comments get displayed. Preview of the old text as it would be displayed in the wiki:
![wikitext](https://user-images.githubusercontent.com/74006324/105201966-bf92c180-5b41-11eb-9e73-0edd0118dcfb.PNG)

> To the BPM diagram changes:
> 
>     * Please add a "data file symbol" behind the compose upload and create backup with a short hint where the files are stored.
Done
>     * I think the provide data in the enhance_data.py section should be remove. Previously it was used to indicate that the data will be passed to the next process. However, this handover no longer exists.
Done

**New Diagram and wiki text in the comments below**

Reply:
**New wiki text:**
## Overview
To populate the Elasticsearch index an ETL batch process has been implemented.
* If data exists, the upload data for the Elasticsearch index gets composed from the backed up data
* The webscrappers scrape the portals from where the data should be acquired
* Then the crawled data gets processed
  * New tags are being searched in the newly acquired data. These tags can be used to expand the current tag ontology
  * The tags then are being ranked according to their absolute (term) frequency
* A backup of the enhanced data is created


## Composing upload data for Elasticsearch
For the composition of the upload, only data from `data_management\backups` is eligible. The data in the most recent backup as well as the number of older backups set in `DataManager.py` will be considered for upload.
This limit of how many older backups should be considered applies to the following description, whenever older backups are mentioned.

At first, the most recent upload data is considered for upload, but will be checked against the data present in older backups according to the values set in `DataManager.py`. If a file in the most recent backup has less than a certain percentage of posts of the files from an older backup,
the file from the older backup is instead chosen for upload. The difference in posts at which the data of an older backup should be uploaded can be set in `DataManager.py`.
If an older backup contains a file that is not present in the most recent data, it is added to the upload data.

The data that got selected for the upload is saved in `data_management\upload` and can be read from there by the process for the upload to Elasticsearch.
A logfile with the timestamp of the creation of the upload data composition can be found under `data_management\upload_data_origin.log`.
In addition to the time of the creation, it also lists which backup each file is chosen from.

Reply:
**New diagram:**
![einanderhelfen_etl_bpmn2](https://user-images.githubusercontent.com/74006324/105206146-53668c80-5b46-11eb-96b1-8414b41216c6.png)